test/lib/testMayaUsdCacheId.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2020 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import unittest
+import os
+import tempfile
+
+import pxr
+from pxr import Usd, UsdGeom, UsdUtils
+
+from maya import cmds
+from mayaUsd import lib as mayaUsdLib
+
+
+class MayaUsdPythonCacheIdTestCase(unittest.TestCase):
+    """
+    Test cases for Cache Id
+    """
+    def __init__(self, *args, **kwargs):
+        super(MayaUsdPythonCacheIdTestCase, self).__init__(*args, **kwargs)
+
+        self.filePath = None
+        self.SC = pxr.UsdUtils.StageCache.Get()
+
+    def setUp(self):
+
+        cmds.file(force=True, new=True)
+        cmds.loadPlugin("mayaUsdPlugin", quiet=True)
+        self.assertTrue(
+            cmds.pluginInfo("mayaUsdPlugin", query=True, loaded=True))
+
+        tmpfile = tempfile.NamedTemporaryFile(delete=True, suffix=".usda")
```

Comment:
I'm confused.  The documentation to NamedTemporaryFile() says that the file is deleted as soon as it's closed, i.e. at line 48.  Are you making this call just to get a temporary file name?

If so, you might want to have a look at the test fixtures code we've been using
https://github.com/Autodesk/maya-usd/blob/dev/test/testUtils/fixturesUtils.py
which places test output in a known directory, where you can inspect it after the fact, and cleans out this directory in subsequent runs.

Reply:
I've used that NamedTemporaryFile trick in the layer command tests, and it is used in a few places in the AL plugin tests, testLayerManager.py, testProxyShape.py and testTranslators.py

Reply:
Yup, seems to work well but I can switch it if you feel strongly about it.

Reply:
AL had issues with NamedTemporaryFile in the past (https://github.com/Autodesk/maya-usd/pull/273#issuecomment-589585095). I don't see why we would need to follow this pattern. @alsoussi please switch to using https://github.com/Autodesk/maya-usd/blob/dev/test/testUtils/fixturesUtils.py

Reply:
Done!

Reply:
@kxl-adsk

Reply:
the tests are failing

    Traceback (most recent call last):
      File "/home/S/jenkins/workspace/ecg-mayausd-branch-preflight-python3-linux/build/RelWithDebInfo/test/lib/testMayaUsdCacheId.py", line 47, in setUp
        cmds.file(filePath, force=True, open=True)
    RuntimeError: File not found: "/home/S/jenkins/workspace/ecg-mayausd-branch-preflight-python3-linux/build/RelWithDebInfo/test/lib/MayaUsdPythonCacheIdTestCase/MayaUsdPythonCacheIdTestCase.ma"

(check pre-flight build below, look for "Artifact" link at the top right to download the log)

Reply:
Yup you are right @luceric-adsk I've fixed them @kxl-adsk